whois-api/src/main/java/net/ripe/db/whois/api/httpserver/dos/LookupDoSFilterHolder.java

```diff
@@ -0,0 +1,42 @@
+package net.ripe.db.whois.api.httpserver.dos;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import static jakarta.ws.rs.HttpMethod.GET;
+
+@Component
+public class LookupDoSFilterHolder extends AbstractDoSFilterHolder {
+
+    private static final String MAX_REQUEST_PER_SECOND = "50";
+    private static final String MAX_REQUEST_PER_MS = "" + 10 * 60 * 1_000; // high default, 10 minutes
+
+    public LookupDoSFilterHolder(@Value("${dos.filter.enabled:false}") final boolean dosFilterEnabled,
+                                 @Value("${ipranges.trusted}") final String trustedIpRanges) {
+        super(dosFilterEnabled, trustedIpRanges);
+        
+        this.setFilter(generateWhoisDoSLookupFilter());
+        this.setName("LookupDoSFilter");
```

Comment:
shift them to abstract class and create a abstract method getFilterName and each implementation have this method

Reply:
yes, done